src/superagent-promise-plugin.js

```diff
@@ -3,8 +3,7 @@ function endPromise(req) {
 
   return new _Promise(function (resolve, reject) {
     req.end(function (err, res) {
-      err = err || res.error;
-      if (err) reject(err);
+      if (err || res.error) reject(res);
```

Comment:
Sounds good. Could you also add a test to make sure response is on the error object?